sympy/core/power.py

```diff
@@ -579,9 +580,12 @@ def _check(ct1, ct2, old):
                     pow = as_int(pow)
                     combines = True
                 except ValueError:
-                    combines = Pow._eval_power(
-                        Pow(*old.as_base_exp(), evaluate=False),
-                        pow) is not None
+                    if denom(pow) == 1:
```

Comment:
> I don't know why there is no testcases for rational, Imaginary power.

All the details of whether to combine a base/power with an exponent is in the `Pow._eval_power` routine. I think the integer check (via denom) is a quick test and if it fails then there is a fallback to the full`_eval_power` routine so as not to duplicate logic -- getting rid of lurking logical errors associated with this issue was a long process.